### PR TITLE
DPL-975c targeted nanoseq limber poly metadata via api v2

### PR DESCRIPTION
**Pull request by andrewsparkes**
Closes #https://github.com/sanger/sequencescape/issues/3941

### CODE REVIEW

File: app/sequencescape/sequencescape/api/v2/poly_metadatum.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+# Represents a poly metadatum in Limber via the Sequencescape v2 API
+class Sequencescape::Api::V2::PolyMetadatum < Sequencescape::Api::V2::Base
+  has_one :metadatable, polymorphic: true, shallow_path: true
```
Comment (KatyTaylor) [inline app/sequencescape/sequencescape/api/v2/poly_metadatum.rb:5]: What does `shallow_path` do?

File: app/sequencescape/sequencescape/api/v2/shared/has_poly_metadata.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+# Module for defning shared concerns that can be included in v2 classes to access
```
Comment (KatyTaylor) [inline app/sequencescape/sequencescape/api/v2/shared/has_poly_metadata.rb:3]:
```suggestion
# Module for defining shared concerns that can be included in v2 classes to access
```

File: spec/sequencescape/api/v2/request_spec.rb [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    # 1. select our metadatable request. e.g.
+    #    r = Sequencescape::Api::V2::Request.find(11180).first
+    # 2. create the new poly metadatum, e.g.
+    #    pm1 = Sequencescape::Api::V2::PolyMetadatum.new(key: 'test_key', value: 'test_value')
+    # 3. set the metadatable on the poly metadatum, e.g.
+    #    pm1.relationships.metadatable = r
+    # 4. save the poly metadatum, e.g.
+    #    pm1.save
+    # If we set the metadatable into the initializer, the save fails as the metadatable is treated as
+    # an attribute and not a relationship.
```
Comment (KatyTaylor) [inline spec/sequencescape/api/v2/request_spec.rb:20]: Is it worth adding this documentation to the v2 /poly_metadatum file as well, to maximise the chance someone who wants to use it sees how to do it?

Comment (andrewsparkes) [reply spec/sequencescape/api/v2/request_spec.rb:20]: Done

File: app/sequencescape/sequencescape/api/v2/poly_metadatum.rb (hunk 1 above)
Comment (KatyTaylor) [reply app/sequencescape/sequencescape/api/v2/poly_metadatum.rb:17]: Oh interesting OK, thanks.

**Review by KatyTaylor — APPROVED** [review]

**Pull request merged**